src/bpfprogram.h

```diff
@@ -35,6 +36,8 @@ class BpfProgram {
 
 private:
   struct bpf_program *bpf_prog_;
+  // This fd is specifically for attaching to other running BPF programs.
+  util::FD attach_fd_ = util::FD(-1);
```

Comment:
The `-1` here is a leaky abstraction (knowledge of the internal `FD` implementation). Just make this type `std::optional<FD> attach_fd_` and then there is no valid state. You can just do things like:

```
attach_fd_.emplace(fd);
...
attach_fd_ ? attach_fd_->get() : 0 /* libbpf weirdness */
```

Reply:
Done.